src/controllers/wallet/atomic-swap/tx-proposal.controller.js

```diff
@@ -335,6 +376,39 @@ async function listenedProposalList(req, res) {
   res.send({ success: true, proposals: list });
 }
 
+/**
+ * Registers a proposal on this wallet, storing its identifier and password locally
+ */
+async function registerProposal(req, res) {
+  const { walletId } = req;
+  const { proposalId } = req.params;
+  const { password } = req.body;
+
+  // Avoid making requests to the service if the proposal is already on local storage
+  const proposalMap = await atomicSwapService.getListenedProposals(walletId);
+  if (proposalMap.has(req.params.proposalId)) {
+    res.send({ success: true });
```

Comment:
Should we validate if the `password` is the same as the stored one? Maybe return success false in case it's different. "You already have a proposal with this ID registered but with a different password."

Reply:
Good idea. Implemented it and refactored the tests on [b123fad](https://github.com/HathorNetwork/hathor-wallet-headless/pull/277/commits/b123fad0010767cbf318d6f310bf621a6afd758e).